pkg/bindings/containers/exec.go

```diff
@@ -114,10 +114,15 @@ func ExecStart(ctx context.Context, sessionID string, options *ExecStartOptions)
 
 // ExecRemove removes a given exec session.
 func ExecRemove(ctx context.Context, sessionID string, options *ExecRemoveOptions) error {
+	v := bindings.ServiceVersion(ctx)
+	// The exec remove endpoint was added in 4.8.
+	if v.Major < 4 || (v.Major == 4 && v.Minor < 8) {
+		// Do no call this endpoint as it will not be supported on the server and throw an "NOT FOUND" error.
+		return nil
+	}
```

Comment:
Drive-by-comment:

I feel this should be handled more on the caller side.

A user may accidentally use an older server version than intended.  In this case, the user may assume that there are no leaking exec sessions anymore as it didn't receive an error.

Reply:
Well the version is always based on the server so the client has no choice. I do not see why any client would like to get a bogus error here. If anything it leads to even more bug reports for us.

Either way I find it broken to rely on the client to do delete the exec session as mentioned on https://github.com/containers/podman/pull/20405. This approach is way to fragile, no docker API user knows that such endpoint exists.

Reply:
I was probably too ambiguous on what I meant by user.

I am in favor of adding such a conditional check to podman-remote.  But I think the caller of the bindings should handle it.

Another external user of the bindings may be surprised about that check as it can hide the fact the client is (mistakenly) talking to a too old server.

Reply:
Sure but retuning the current error is just bogus and a normal user would not be able to interpret such error correctly.

Reply:
pushed a second commit to return a custom error type that callers can match for.
This should make it future proof and all new bindings function should then throw this error if the server is to old.